src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/EcmaFormat/ReflectionTypeProvider.cs

```diff
@@ -0,0 +1,306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Runtime.InteropServices;
+using System.Reflection.Metadata;
+using System.Reflection.Runtime.General;
+using System.Reflection.Runtime.TypeInfos;
+
+namespace System.Reflection.Runtime
+{
+    static class ReflectionTypeProviderHelpers
+    {
+        private static Type s_objectType = typeof(object);
+        private static Type s_intPtrType = typeof(IntPtr);
+        private static Type s_int64Type = typeof(long);
+        private static Type s_int32Type = typeof(int);
+        private static Type s_int16Type = typeof(short);
+        private static Type s_int8Type = typeof(sbyte);
+        private static Type s_uintPtrType = typeof(UIntPtr);
+        private static Type s_uint64Type = typeof(ulong);
+        private static Type s_uint32Type = typeof(uint);
+        private static Type s_uint16Type = typeof(ushort);
+        private static Type s_uint8Type = typeof(byte);
+        private static Type s_charType = typeof(char);
+        private static Type s_floatType = typeof(float);
+        private static Type s_doubleType = typeof(double);
+        private static Type s_boolType = typeof(bool);
+        private static Type s_stringType = typeof(string);
+        private static Type s_voidType = typeof(void);
```

Comment:
We already import CommonRuntimeTypes.cs into Reflection\Core - that should let you rid of these private caches.